examples/rcbd_one_var.R

```diff
@@ -0,0 +1,41 @@
+wheat.data <- read.csv("Wheat_yield_data.csv", header = TRUE, stringsAsFactors = FALSE)
+
+library(car)  # for leveneTest()
+library(agricolae)  # for LSD.test()
+
+#Inform R that Block and Treatment are factors
+wheat.data$Block<-as.factor(wheat.data$Block)
+wheat.data$Treatment<-as.factor(wheat.data$Treatment)
+
+##The ANOVA##
+wheat.mod<-lm(Yield ~ Treatment + Block, wheat.data)
+anova(wheat.mod)
+
+##TESTING ASSUMPTIONS##
+#Generate residual and predicted values
+wheat.data$resids <- residuals(wheat.mod)
+wheat.data$preds <- predict(wheat.mod)
+wheat.data$sq_preds <- wheat.data$preds^2
+
+#Look at a plot of residual vs. predicted values
+plot(resids ~ preds, data = wheat.data,
+     xlab = "Predicted Values", ylab = "Residuals")
+#Create Box Plot of treatments
+boxplot(Yield ~ Treatment, data = wheat.data,
+        main = "Effect of wheat variety on yield",
+        xlab = "Wheat cultivar", ylab = "Yield (ton/ha)")
+#Perform a Shapiro-Wilk test for normality of residuals
+shapiro.test(wheat.data$resids)
+qqnorm(resid(wheat.mod))
+plot(density(resid(wheat.mod)))
```

Comment:
I would note that the Q-Q plot is much better at determining (by eye) if your residuals are normal. It plots the distribution of your plot against a normal distribution of the same mean and variance so all you have to do is see whether you have a line or not instead of guessing if the distribution is bell shaped.